### PR TITLE
Sc 37 register user addresses

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/flareContract.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    chainId: config.networkID,
+    gasPrice,
+    gasLimit: gasEstimate
+  }
+
+  const serializedUnsignedTx = ethers.utils.serializeTransaction(unsignedTx);
+  const txHash = ethers.utils.keccak256(serializedUnsignedTx);
+  const unsignedTxObj = createUnsignedJsonObject(txHash)
+
+  if (wallet == "Ledger") {
```
Comment (avisingh2) [inline src/flareContract.ts]: you already have defined the Ledger , ForDefi, and private key as constant right. Please use those.

File: src/flareContract.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    if (!derivationPath) throw new Error("No derivation path passed")
+    const sign = await ledgerSign(unsignedTxObj, derivationPath)
+    const serializedSignedTx = ethers.utils.serializeTransaction(unsignedTx, prefix0x(sign.signature))
+    await contract.provider.sendTransaction(serializedSignedTx)
+  }
+  else if (wallet == "ForDefi") {
```
Comment (avisingh2) [inline src/flareContract.ts]: same here

File: src/flareContract.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    if (!transactionId) throw new Error("No transaction Id passed")
+    saveUnsignedTxJson(unsignedTxObj, transactionId)
+    saveUnsignedEVMObject(unsignedTx, transactionId)
+  }
+  else if (wallet == "Private Key") {
```
Comment (avisingh2) [inline src/flareContract.ts]: same here

File: src/cli.ts [hunk 4]
```diff
@@ -408,7 +409,13 @@ async function cliSendSignedTxJson(ctx: Context, id: string) {
   if (isAlreadySentToChain(id)) {
     throw new Error("Tx already sent to chain")
   }
-  const chainTxId = await sendSignedTxJson(ctx, readSignedTxJson(id))
+  const signedTxnJson = readSignedTxJson(id)
+  let chainTxId
+  if (signedTxnJson.transactionType === "RegisterAddress") {
```
Comment (avisingh2) [inline src/cli.ts]: if you have defined the `RegisterAddress` as const please use that

File: src/forDefi/forDefi.ts [hunk 5]
```diff
@@ -57,7 +57,7 @@ export async function sendToForDefi(unsignedTxidFile: string, ctxFile: string, w
     const timestamp = new Date().getTime();
     const payload = `${path}|${timestamp}|${requestBody}`;
 
-    const privateKeyFile = "../private.pem"
+    const privateKeyFile = "./private.pem"
```
Comment (avisingh2) [inline src/forDefi/forDefi.ts:60]: not sure but is it correct to store the private key in cli?

Comment (ghost) [reply src/forDefi/forDefi.ts:60]: This change was requested by the flare team. they wanted us to change the directory from the parent dir to the current dir for both token and pem file.

**Review by avisingh2 — APPROVED** [review]

**Pull request merged**